tests/unit_tests/test_source.py

```diff
@@ -134,3 +135,29 @@ def test_rejection(run_in_tmpdir):
         assert p.r not in non_source_region
 
     openmc.lib.finalize()
+
+
+def test_exceptions():
+
+    with pytest.raises(AttributeError, match=r'Please use the FileSource class'):
+        s = openmc.IndependentSource(filename='my_file')
+
+    with pytest.raises(AttributeError, match=r'Please use the CompiledSource class'):
+        s = openmc.IndependentSource(library='my_library')
+
+    with pytest.raises(AttributeError, match=r'Please use the CompiledSource class'):
+        s = openmc.IndependentSource(parameters='my_params')
+
+    with pytest.raises(AttributeError, match=r'Please use the FileSource class'):
+        s = openmc.IndependentSource()
+        s.file = 'my_file'
+
+    with pytest.raises(AttributeError, match=r'Please use the CompiledSource class'):
+        s = openmc.IndependentSource(library='my_library')
+        s.library = 'my_library'
+
+    with pytest.raises(AttributeError, match=r'Please use the CompiledSource class'):
+        s = openmc.IndependentSource(parameters='my_params')
+        s.parameters = 'my_params'
```

Comment:
The second line in each of these `with` blocks won't be checked

Reply:
Ah of course. Thanks!